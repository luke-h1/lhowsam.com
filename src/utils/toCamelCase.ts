export default function toCamelCase(str: string) {
  return str
    .split(' ')
    .map((word, index) => {
      if (index === 0) {
        return word;
      }
      return word.charAt(0) + word.slice(1);
    })
    .join(''); // Join all the words back together
}
