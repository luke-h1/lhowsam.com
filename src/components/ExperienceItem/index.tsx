import * as Grid from '@frontend/components/Grid';
import * as List from '@frontend/components/List';
import { Company } from '@frontend/config/jobs';
import { variables } from '@frontend/styles/variables.css';
import { parseDate } from '@frontend/utils/date';
import toCamelCase from '@frontend/utils/toCamelCase';
import { format } from 'date-fns';
import Box from '../Box';
import Heading from '../Heading';
import Text from '../Text';

interface Props {
  company: Company;
}

export default function ExperienceItem({ company }: Props) {
  return (
    <Box
      key={company.title}
      style={{
        borderBottom: `1px solid ${variables.color.border}`,
        marginBottom: variables.spacing.lg,
      }}
    >
      <Heading fontSize="md" style={{ marginBottom: variables.spacing.sm }}>
        {company.title}
      </Heading>
      <List.Container
        testId={`ExperienceItem-list-${toCamelCase(company.title)}`}
      >
        {company.jobs
          .sort(
            (a, b) =>
              parseDate(b.startDate).getTime() -
              parseDate(a.startDate).getTime(),
          )
          .map(job => (
            <List.Item key={job.id}>
              <Grid.Container rowGap="md" alignItems="baseline">
                <Grid.Column
                  colStart={{ xs: '1' }}
                  colEnd={{ xs: '-1', md: '1' }}
                >
                  <Heading as="h3" color="foregroundNeutral">
                    {job.title}
                  </Heading>
                  <Text color="foregroundNeutral" fontSize="sm">
                    {job.location}
                  </Text>
                </Grid.Column>
                <Grid.Column
                  colStart={{ xs: '1', md: '2' }}
                  colEnd={{ xs: '-1', md: '4' }}
                >
                  <Text
                    testId={`${toCamelCase(company.title)}-${toCamelCase(job.title)}-description`}
                  >
                    {job.description || 'No description yet'}
                  </Text>
                </Grid.Column>
                <Grid.Column
                  colStart={{ xs: '1', md: '4' }}
                  colEnd={{ xs: '-1', md: '4' }}
                >
                  <Text
                    color="foregroundNeutral"
                    fontSize="sm"
                    testId={`${toCamelCase(company.title)}-${toCamelCase(job.title)}-date-range`}
                  >
                    {format(parseDate(job.startDate), 'MMM yyyy')} &mdash;{' '}
                    {job.endDate
                      ? format(parseDate(job.endDate), 'MMM yyyy')
                      : 'Now'}
                    <br />
                  </Text>
                </Grid.Column>
              </Grid.Container>
            </List.Item>
          ))}
      </List.Container>
    </Box>
  );
}
