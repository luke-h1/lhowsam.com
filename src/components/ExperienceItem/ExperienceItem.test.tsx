import companies, { Company } from '@frontend/config/jobs';
import render from '@frontend/test/render';
import toCamelCase from '@frontend/utils/toCamelCase';
import { screen, within } from '@testing-library/react';
import ExperienceItem from '.';

const floow = companies.find(
  company => company.title === 'The Floow',
) as Company;

const hive = companies.find(company => company.title === 'Hive IT') as Company;

const skyBet = companies.find(
  company => company.title === 'Sky Betting & Gaming',
) as Company;

const flutter = companies.find(
  company => company.title === 'Flutter UKI',
) as Company;

describe('ExperienceItem', () => {
  test.each(companies)('should render title for %s', company => {
    const { container } = render(<ExperienceItem company={company} />);
    expect(screen.getByText(company.title)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  describe('Floow', () => {
    test('renders all jobs', () => {
      render(<ExperienceItem company={floow} />);

      const list = screen.getByTestId('ExperienceItem-list-TheFloow');
      const items = within(list).getAllByRole('listitem');

      expect(items).toHaveLength(1);
    });

    test('renders in desc order', () => {
      render(<ExperienceItem company={floow} />);

      const list = screen.getByTestId('ExperienceItem-list-TheFloow');
      const items = within(list).getAllByRole('listitem');

      expect(items[0]).toHaveTextContent('Junior service desk analyst');
    });

    test('renders correct date range', () => {
      render(<ExperienceItem company={floow} />);

      expect(
        screen.getByTestId('TheFloow-Juniorservicedeskanalyst-date-range'),
      ).toHaveTextContent('Jun 2019 — Jul 2020');
    });

    test.each(floow.jobs)('renders description for %s', job => {
      render(<ExperienceItem company={floow} />);

      expect(
        screen.getByTestId(`TheFloow-${toCamelCase(job.title)}-description`),
      ).toHaveTextContent(job.description as string);
    });
  });

  describe('Hive IT', () => {
    test('renders all jobs', () => {
      render(<ExperienceItem company={hive} />);

      const list = screen.getByTestId('ExperienceItem-list-HiveIT');
      const items = within(list).getAllByRole('listitem');

      expect(items).toHaveLength(3);
    });

    test('renders in desc order', () => {
      render(<ExperienceItem company={hive} />);

      const list = screen.getByTestId('ExperienceItem-list-HiveIT');
      const items = within(list).getAllByRole('listitem');

      expect(items[0]).toHaveTextContent('Software Engineer');
      expect(items[1]).toHaveTextContent('Software Tester');
      expect(items[2]).toHaveTextContent('Junior Software Tester');
    });

    test('renders correct dates', () => {
      render(<ExperienceItem company={hive} />);

      const list = screen.getByTestId('ExperienceItem-list-HiveIT');
      const items = within(list).getAllByRole('listitem');

      expect(items[0]).toHaveTextContent('May 2022 — Aug 2023');
      expect(items[1]).toHaveTextContent('May 2021 — May 2022');
      expect(items[2]).toHaveTextContent('Oct 2020 — May 2021');
    });

    test('renders correct date range', () => {
      render(<ExperienceItem company={hive} />);

      expect(
        screen.getByTestId('HiveIT-SoftwareEngineer-date-range'),
      ).toHaveTextContent('May 2022 — Aug 2023');

      expect(
        screen.getByTestId('HiveIT-SoftwareTester-date-range'),
      ).toHaveTextContent('May 2021 — May 2022');

      expect(
        screen.getByTestId('HiveIT-JuniorSoftwareTester-date-range'),
      ).toHaveTextContent('Oct 2020 — May 2021');
    });

    test.each(hive.jobs)('renders description for %s', job => {
      render(<ExperienceItem company={hive} />);

      expect(
        screen.getByTestId(`HiveIT-${toCamelCase(job.title)}-description`),
      ).toHaveTextContent(job.description as string);
    });
  });

  describe('SkyBet', () => {
    test('renders all jobs', () => {
      render(<ExperienceItem company={skyBet} />);

      const list = screen.getByTestId('ExperienceItem-list-SkyBetting&Gaming');
      const items = within(list).getAllByRole('listitem');
      expect(items).toHaveLength(1);

      expect(items[0]).toHaveTextContent('Software Engineer');
    });

    test('renders correct date range', () => {
      render(<ExperienceItem company={skyBet} />);

      const list = screen.getByTestId('ExperienceItem-list-SkyBetting&Gaming');
      const items = within(list).getAllByRole('listitem');

      expect(items[0]).toHaveTextContent('Aug 2023 — Nov 2024');
    });

    test('renders correct date range', () => {
      render(<ExperienceItem company={skyBet} />);

      expect(
        screen.getByTestId('SkyBetting&Gaming-SoftwareEngineer-date-range'),
      ).toHaveTextContent('Aug 2023 — Nov 2024');
    });

    test('renders description', () => {
      render(<ExperienceItem company={skyBet} />);

      expect(
        screen.getByTestId('SkyBetting&Gaming-SoftwareEngineer-description'),
      ).toHaveTextContent(skyBet.jobs[0].description as string);
    });
  });

  describe('Flutter', () => {
    test('renders all jobs', () => {
      render(<ExperienceItem company={flutter} />);
      const list = screen.getByTestId('ExperienceItem-list-FlutterUKI');
      const items = within(list).getAllByRole('listitem');
      expect(items).toHaveLength(1);

      expect(items[0]).toHaveTextContent('Software Developer');
    });

    test('renders correct date range', () => {
      render(<ExperienceItem company={flutter} />);

      const list = screen.getByTestId('ExperienceItem-list-FlutterUKI');
      const items = within(list).getAllByRole('listitem');

      expect(items[0]).toHaveTextContent('Jan 2025 — Now');
    });

    test('renders description', () => {
      render(<ExperienceItem company={flutter} />);

      expect(
        screen.getByTestId('FlutterUKI-SoftwareDeveloper-description'),
      ).toHaveTextContent(flutter.jobs[0].description as string);
    });
  });
});
