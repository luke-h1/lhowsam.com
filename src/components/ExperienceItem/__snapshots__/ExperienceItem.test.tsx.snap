// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperienceItem should render title for { id: 1, title: 'Hive IT', jobs: [Array] } 1`] = `
<div>
  <script
    nonce=""
  >
    ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div
    class=""
  >
    <h2
      class="Heading_root__xkw1cf0 Heading_tracking_h2__xkw1cf2 sprinkles_fontSize_md_xs__1fx8prsj8 sprinkles_color_foreground__1fx8prsm5"
    >
      Hive IT
    </h2>
    <ul
      class="List_list__3nqhft0"
      data-testid="ExperienceItem-list-HiveIT"
    >
      <li
        class="List_item__3nqhft1"
      >
        <div
          class="Grid_container__1otflmc0 sprinkles_columnGap_xxxl_xs__1fx8prsfo sprinkles_rowGap_md_xs__1fx8prse4 sprinkles_alignItems_baseline_xs__1fx8prs2s"
        >
          <div
            class="sprinkles_gridColumnStart_1_xs__1fx8prsjw sprinkles_gridColumnEnd_-1_xs__1fx8prskw sprinkles_gridColumnEnd_1_md__1fx8prski"
          >
            <h3
              class="Heading_root__xkw1cf0 Heading_tracking_h3__xkw1cf3 sprinkles_fontSize_md_xs__1fx8prsj8 sprinkles_color_foregroundNeutral__1fx8prsm6"
            >
              Software Engineer
            </h3>
            <p
              class="sprinkles_maxWidth_prose_xs__1fx8prsfw sprinkles_fontSize_sm_xs__1fx8prsj4 sprinkles_fontWeight_normal__1fx8prsn0 sprinkles_color_foregroundNeutral__1fx8prsm6"
            >
              Sheffield, UK
            </p>
          </div>
          <div
            class="sprinkles_gridColumnStart_1_xs__1fx8prsjw sprinkles_gridColumnStart_2_md__1fx8prsk2 sprinkles_gridColumnEnd_-1_xs__1fx8prskw sprinkles_gridColumnEnd_4_md__1fx8prsku"
          >
            <p
              class="sprinkles_maxWidth_prose_xs__1fx8prsfw sprinkles_fontSize_md_xs__1fx8prsj8 sprinkles_fontWeight_normal__1fx8prsn0 sprinkles_color_foreground__1fx8prsm5"
              data-testid="HiveIT-SoftwareEngineer-description"
            >
              Worked as a full-stack software engineer at a local digital agency mainly on government and charity projects. Responsible for driving DevOps + internal process improvement, feature + maintenance tasks ,presenting technical proof of concepts to stakeholders and supporting + mentoring teammates
            </p>
          </div>
          <div
            class="sprinkles_gridColumnStart_1_xs__1fx8prsjw sprinkles_gridColumnStart_4_md__1fx8prska sprinkles_gridColumnEnd_-1_xs__1fx8prskw sprinkles_gridColumnEnd_4_md__1fx8prsku"
          >
            <p
              class="sprinkles_maxWidth_prose_xs__1fx8prsfw sprinkles_fontSize_sm_xs__1fx8prsj4 sprinkles_fontWeight_normal__1fx8prsn0 sprinkles_color_foregroundNeutral__1fx8prsm6"
              data-testid="HiveIT-SoftwareEngineer-date-range"
            >
              May 2022
               —
               
              Aug 2023
              <br />
            </p>
          </div>
        </div>
      </li>
      <li
        class="List_item__3nqhft1"
      >
        <div
          class="Grid_container__1otflmc0 sprinkles_columnGap_xxxl_xs__1fx8prsfo sprinkles_rowGap_md_xs__1fx8prse4 sprinkles_alignItems_baseline_xs__1fx8prs2s"
        >
          <div
            class="sprinkles_gridColumnStart_1_xs__1fx8prsjw sprinkles_gridColumnEnd_-1_xs__1fx8prskw sprinkles_gridColumnEnd_1_md__1fx8prski"
          >
            <h3
              class="Heading_root__xkw1cf0 Heading_tracking_h3__xkw1cf3 sprinkles_fontSize_md_xs__1fx8prsj8 sprinkles_color_foregroundNeutral__1fx8prsm6"
            >
              Software Tester
            </h3>
            <p
              class="sprinkles_maxWidth_prose_xs__1fx8prsfw sprinkles_fontSize_sm_xs__1fx8prsj4 sprinkles_fontWeight_normal__1fx8prsn0 sprinkles_color_foregroundNeutral__1fx8prsm6"
            >
              Sheffield, UK
            </p>
          </div>
          <div
            class="sprinkles_gridColumnStart_1_xs__1fx8prsjw sprinkles_gridColumnStart_2_md__1fx8prsk2 sprinkles_gridColumnEnd_-1_xs__1fx8prskw sprinkles_gridColumnEnd_4_md__1fx8prsku"
          >
            <p
              class="sprinkles_maxWidth_prose_xs__1fx8prsfw sprinkles_fontSize_md_xs__1fx8prsj8 sprinkles_fontWeight_normal__1fx8prsn0 sprinkles_color_foreground__1fx8prsm5"
              data-testid="HiveIT-SoftwareTester-description"
            >
              Worked on improving the existing testing culture within the company to reduce technical debt and improve happiness
            </p>
          </div>
          <div
            class="sprinkles_gridColumnStart_1_xs__1fx8prsjw sprinkles_gridColumnStart_4_md__1fx8prska sprinkles_gridColumnEnd_-1_xs__1fx8prskw sprinkles_gridColumnEnd_4_md__1fx8prsku"
          >
            <p
              class="sprinkles_maxWidth_prose_xs__1fx8prsfw sprinkles_fontSize_sm_xs__1fx8prsj4 sprinkles_fontWeight_normal__1fx8prsn0 sprinkles_color_foregroundNeutral__1fx8prsm6"
              data-testid="HiveIT-SoftwareTester-date-range"
            >
              May 2021
               —
               
              May 2022
              <br />
            </p>
          </div>
        </div>
      </li>
      <li
        class="List_item__3nqhft1"
      >
        <div
          class="Grid_container__1otflmc0 sprinkles_columnGap_xxxl_xs__1fx8prsfo sprinkles_rowGap_md_xs__1fx8prse4 sprinkles_alignItems_baseline_xs__1fx8prs2s"
        >
          <div
            class="sprinkles_gridColumnStart_1_xs__1fx8prsjw sprinkles_gridColumnEnd_-1_xs__1fx8prskw sprinkles_gridColumnEnd_1_md__1fx8prski"
          >
            <h3
              class="Heading_root__xkw1cf0 Heading_tracking_h3__xkw1cf3 sprinkles_fontSize_md_xs__1fx8prsj8 sprinkles_color_foregroundNeutral__1fx8prsm6"
            >
              Junior Software Tester
            </h3>
            <p
              class="sprinkles_maxWidth_prose_xs__1fx8prsfw sprinkles_fontSize_sm_xs__1fx8prsj4 sprinkles_fontWeight_normal__1fx8prsn0 sprinkles_color_foregroundNeutral__1fx8prsm6"
            >
              Sheffield, UK
            </p>
          </div>
          <div
            class="sprinkles_gridColumnStart_1_xs__1fx8prsjw sprinkles_gridColumnStart_2_md__1fx8prsk2 sprinkles_gridColumnEnd_-1_xs__1fx8prskw sprinkles_gridColumnEnd_4_md__1fx8prsku"
          >
            <p
              class="sprinkles_maxWidth_prose_xs__1fx8prsfw sprinkles_fontSize_md_xs__1fx8prsj8 sprinkles_fontWeight_normal__1fx8prsn0 sprinkles_color_foreground__1fx8prsm5"
              data-testid="HiveIT-JuniorSoftwareTester-description"
            >
              Worked as a Junior software tester to take over from the previous tester. Built up testing skills both automated and manual to aid in the maintenance of large and small projects
            </p>
          </div>
          <div
            class="sprinkles_gridColumnStart_1_xs__1fx8prsjw sprinkles_gridColumnStart_4_md__1fx8prska sprinkles_gridColumnEnd_-1_xs__1fx8prskw sprinkles_gridColumnEnd_4_md__1fx8prsku"
          >
            <p
              class="sprinkles_maxWidth_prose_xs__1fx8prsfw sprinkles_fontSize_sm_xs__1fx8prsj4 sprinkles_fontWeight_normal__1fx8prsn0 sprinkles_color_foregroundNeutral__1fx8prsm6"
              data-testid="HiveIT-JuniorSoftwareTester-date-range"
            >
              Oct 2020
               —
               
              May 2021
              <br />
            </p>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`ExperienceItem should render title for { id: 2, title: 'Sky Betting & Gaming', jobs: [Array] } 1`] = `
<div>
  <script
    nonce=""
  >
    ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div
    class=""
  >
    <h2
      class="Heading_root__xkw1cf0 Heading_tracking_h2__xkw1cf2 sprinkles_fontSize_md_xs__1fx8prsj8 sprinkles_color_foreground__1fx8prsm5"
    >
      Sky Betting & Gaming
    </h2>
    <ul
      class="List_list__3nqhft0"
      data-testid="ExperienceItem-list-SkyBetting&Gaming"
    >
      <li
        class="List_item__3nqhft1"
      >
        <div
          class="Grid_container__1otflmc0 sprinkles_columnGap_xxxl_xs__1fx8prsfo sprinkles_rowGap_md_xs__1fx8prse4 sprinkles_alignItems_baseline_xs__1fx8prs2s"
        >
          <div
            class="sprinkles_gridColumnStart_1_xs__1fx8prsjw sprinkles_gridColumnEnd_-1_xs__1fx8prskw sprinkles_gridColumnEnd_1_md__1fx8prski"
          >
            <h3
              class="Heading_root__xkw1cf0 Heading_tracking_h3__xkw1cf3 sprinkles_fontSize_md_xs__1fx8prsj8 sprinkles_color_foregroundNeutral__1fx8prsm6"
            >
              Software Engineer
            </h3>
            <p
              class="sprinkles_maxWidth_prose_xs__1fx8prsfw sprinkles_fontSize_sm_xs__1fx8prsj4 sprinkles_fontWeight_normal__1fx8prsn0 sprinkles_color_foregroundNeutral__1fx8prsm6"
            >
              Leeds, UK
            </p>
          </div>
          <div
            class="sprinkles_gridColumnStart_1_xs__1fx8prsjw sprinkles_gridColumnStart_2_md__1fx8prsk2 sprinkles_gridColumnEnd_-1_xs__1fx8prskw sprinkles_gridColumnEnd_4_md__1fx8prsku"
          >
            <p
              class="sprinkles_maxWidth_prose_xs__1fx8prsfw sprinkles_fontSize_md_xs__1fx8prsj8 sprinkles_fontWeight_normal__1fx8prsn0 sprinkles_color_foreground__1fx8prsm5"
              data-testid="SkyBetting&Gaming-SoftwareEngineer-description"
            >
              Worked within the GLS (Game Launch Service) squad that provided game-launching capabilities for all gaming content across the business in addition to Reality Check (a safer gambling tool). Responsible for maintaining internal NPM packages, shared SDKs, web apps, serverless functions, driving DevOps improvements, monitoring our services (NewRelic, Grafana & Prometheus), supporting + mentoring teammates and captain duties (support queries etc.)
            </p>
          </div>
          <div
            class="sprinkles_gridColumnStart_1_xs__1fx8prsjw sprinkles_gridColumnStart_4_md__1fx8prska sprinkles_gridColumnEnd_-1_xs__1fx8prskw sprinkles_gridColumnEnd_4_md__1fx8prsku"
          >
            <p
              class="sprinkles_maxWidth_prose_xs__1fx8prsfw sprinkles_fontSize_sm_xs__1fx8prsj4 sprinkles_fontWeight_normal__1fx8prsn0 sprinkles_color_foregroundNeutral__1fx8prsm6"
              data-testid="SkyBetting&Gaming-SoftwareEngineer-date-range"
            >
              Aug 2023
               —
               
              Nov 2024
              <br />
            </p>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`ExperienceItem should render title for { id: 3, title: 'Flutter UKI', jobs: [Array] } 1`] = `
<div>
  <script
    nonce=""
  >
    ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div
    class=""
  >
    <h2
      class="Heading_root__xkw1cf0 Heading_tracking_h2__xkw1cf2 sprinkles_fontSize_md_xs__1fx8prsj8 sprinkles_color_foreground__1fx8prsm5"
    >
      Flutter UKI
    </h2>
    <ul
      class="List_list__3nqhft0"
      data-testid="ExperienceItem-list-FlutterUKI"
    >
      <li
        class="List_item__3nqhft1"
      >
        <div
          class="Grid_container__1otflmc0 sprinkles_columnGap_xxxl_xs__1fx8prsfo sprinkles_rowGap_md_xs__1fx8prse4 sprinkles_alignItems_baseline_xs__1fx8prs2s"
        >
          <div
            class="sprinkles_gridColumnStart_1_xs__1fx8prsjw sprinkles_gridColumnEnd_-1_xs__1fx8prskw sprinkles_gridColumnEnd_1_md__1fx8prski"
          >
            <h3
              class="Heading_root__xkw1cf0 Heading_tracking_h3__xkw1cf3 sprinkles_fontSize_md_xs__1fx8prsj8 sprinkles_color_foregroundNeutral__1fx8prsm6"
            >
              Software Developer
            </h3>
            <p
              class="sprinkles_maxWidth_prose_xs__1fx8prsfw sprinkles_fontSize_sm_xs__1fx8prsj4 sprinkles_fontWeight_normal__1fx8prsn0 sprinkles_color_foregroundNeutral__1fx8prsm6"
            >
              Leeds, UK
            </p>
          </div>
          <div
            class="sprinkles_gridColumnStart_1_xs__1fx8prsjw sprinkles_gridColumnStart_2_md__1fx8prsk2 sprinkles_gridColumnEnd_-1_xs__1fx8prskw sprinkles_gridColumnEnd_4_md__1fx8prsku"
          >
            <p
              class="sprinkles_maxWidth_prose_xs__1fx8prsfw sprinkles_fontSize_md_xs__1fx8prsj8 sprinkles_fontWeight_normal__1fx8prsn0 sprinkles_color_foreground__1fx8prsm5"
              data-testid="FlutterUKI-SoftwareDeveloper-description"
            >
              Greenfield React Native development
            </p>
          </div>
          <div
            class="sprinkles_gridColumnStart_1_xs__1fx8prsjw sprinkles_gridColumnStart_4_md__1fx8prska sprinkles_gridColumnEnd_-1_xs__1fx8prskw sprinkles_gridColumnEnd_4_md__1fx8prsku"
          >
            <p
              class="sprinkles_maxWidth_prose_xs__1fx8prsfw sprinkles_fontSize_sm_xs__1fx8prsj4 sprinkles_fontWeight_normal__1fx8prsn0 sprinkles_color_foregroundNeutral__1fx8prsm6"
              data-testid="FlutterUKI-SoftwareDeveloper-date-range"
            >
              Jan 2025
               —
               
              Now
              <br />
            </p>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
